bigquery/google/cloud/bigquery/magics.py

```diff
@@ -0,0 +1,251 @@
+
+
+context = Context()
+
+
+def _run_query(client, query, job_config=None):
+    """Runs a query while printing status updates
+
+    Args:
+        client (google.cloud.bigquery.client.Client):
+            Client to bundle configuration needed for API requests.
+        query (str):
+            SQL query to be executed. Defaults to the standard SQL dialect.
+            Use the ``job_config`` parameter to change dialects.
+        job_config (google.cloud.bigquery.job.QueryJobConfig, optional):
+            Extra configuration options for the job.
+
+    Returns:
+        str: the ID of the query job created
+
+    Example:
+        >>> client = bigquery.Client()
+        >>> _run_query(client, "SELECT 17")
+        Executing query with job ID: bf633912-af2c-4780-b568-5d868058632b
+        Query executing: 1.66s
+        Query complete after 2.07s
+        'bf633912-af2c-4780-b568-5d868058632b'
+    """
+    start_time = time.time()
+    query_job = client.query(query, job_config=job_config)
+    print('Executing query with job ID: {}'.format(query_job.job_id))
+
+    while True:
+        print('\rQuery executing: {:0.2f}s'.format(
+            time.time() - start_time), end='')
+        try:
+            query_job.result(timeout=0.5)
+        except futures.TimeoutError:
+            continue
+        break
+    print('\nQuery complete after {:0.2f}s'.format(time.time() - start_time))
+    return query_job.job_id
+
+
+@magic_arguments.magic_arguments()
+@magic_arguments.argument(
+    'destination_var',
+    nargs='?',
+    help=('If provided, save the output to this variable in addition '
+          'to displaying it.'))
+@magic_arguments.argument(
+    '--project',
+    type=str,
+    default=None,
+    help=('Project to use for executing this query. Defaults to the '
+          'client\'s project'))
+@magic_arguments.argument(
+    '--use_legacy_sql', action='store_true', default=False,
+    help=('Sets query to use Legacy SQL instead of Standard SQL. Defaults to '
+          'Standard SQL if this argument is not used.'))
+@magic_arguments.argument(
+    '--verbose', action='store_true', default=False,
+    help=('If set, print verbose output, including the query job ID and the '
+          'amount of time for the query to finish. By default, this '
+          'information will be displayed as the query runs, but will be '
+          'cleared after the query is finished.'))
+def _cell_magic(line, query):
+    """Underlying function for bigquery cell magic
+
+        Note:
+            This function contains the underlying logic for the 'bigquery' cell
+            magic. This function is not meant to be called directly.
+
+        Args:
+            line (str): "%%bigquery" followed by arguments as required
+            query (str): SQL query to run
+
+        Returns:
+            pandas.DataFrame: the query results.
+    """
+    args = magic_arguments.parse_argstring(_cell_magic, line)
+
+    if args.project:
+        client = Client(
```

Comment:
Does the query function not take a project argument? Should we add a feature request to allow that (likewise with the other functions to create jobs)?